### PR TITLE
Add postgres bulk insertion mode for tracing

### PR DESCRIPTION
**Pull request by efritz**
For postgres bulk inserts, we cut out the placeholders (which doesn't hold any meaningful data) and do not list every argument separately, but give a count of arguments. This should stop errors during upload processing about oversized jaeger spans.

### CODE REVIEW

File: internal/db/dbconn/dbconn.go [hunk 1]
```diff
@@ -166,19 +167,52 @@ func Open(dataSource string) (*sql.DB, error) {
...
+	return context.WithValue(ctx, bulkInsertionKey, bulkInsertion)
+}
+
 type hook struct{}
 
+// postgresBulkInsertRowPattern matches `($1, $2, $3), ($4, $5, $6), ...` which
+// we use to cut out the row payloads from bulk insertion tracing data. We don't
+// need all the parameter data for such requests, which are too big to fit into
+// Jaeger spans.
+var postgresBulkInsertRowPattern = regexp.MustCompile(`(\([$\d,\s]+\)[,\s]*)+`)
```
Comment (keegancsmith) [inline internal/db/dbconn/dbconn.go:194]: nit: use lazyregexp pkg

File: internal/db/dbconn/dbconn.go [hunk 2]
```diff
@@ -166,19 +167,52 @@ func Open(dataSource string) (*sql.DB, error) {
...
+
 // Before implements sqlhooks.Hooks
 func (h *hook) Before(ctx context.Context, query string, args ...interface{}) (context.Context, error) {
+	if BulkInsertion(ctx) {
+		query = string(postgresBulkInsertRowPattern.ReplaceAll([]byte(query), []byte("...")))
```
Comment (keegancsmith) [inline internal/db/dbconn/dbconn.go:199]: For extra protection, can we just have a max length we log on query for all query types? eg if query is longer than 256 chars, truncate it.

Alternatively we could remove the `BulkInsertion` hint, and always do this if query or args is large enough.

**Review by keegancsmith — APPROVED** [review]

**Review by unknwon — APPROVED** [review]

File: internal/db/dbconn/dbconn.go (hunk 1 above)
Comment (efritz) [reply internal/db/dbconn/dbconn.go:194]: I will. Can I ask what the benefit is there?

File: internal/db/dbconn/dbconn.go (hunk 2 above)
Comment (efritz) [reply internal/db/dbconn/dbconn.go:199]: I didn't want to truncate as there might be something useful at the end of the bulk insert that you'd need to dive into source in order to to see (e.g. `INSERT INTO something (a, b, c) VALUES ... ON CONFLICT DO NOTHING`).

I think this is a useful enough thing to keep.

**Pull request merged**